### PR TITLE
boards: stm32: Add support for MXChip AZ3166 IoT DevKit

### PR DESCRIPTION
**Pull request by kartben**
This adds support for the MXChip AZ3166 IoT DevKit, on-board sensors and OLED display, a dev kit that's been widely distributed by Microsoft over the years as their reference devkit for Azure IoT. Note that the board definition in this PR does *not* include support for Wi-Fi (this would require a binary blob and a custom driver).

All sensors have been tested to work properly with their respective samples (+sensor shell) and the OLED display has also been tested.

### CODE REVIEW

File: boards/arm/az3166_iotdevkit/az3166_iotdevkit.dts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		reg = <0x3c>;
+		width = <128>;
+		height = <64>;
+		segment-offset = <0>;
+		page-offset = <0>;
+		display-offset = <0>;
+		multiplex-ratio = <63>;
+		segment-remap;
+		com-invdir;
+		prechargep = <0x22>; // check
```
Comment (erwango) [inline boards/arm/az3166_iotdevkit/az3166_iotdevkit.dts:171]: Nit: `// check` leftover

File: boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+CONFIG_GPIO=y
+
+# Enable Clocks
+CONFIG_CLOCK_CONTROL=y
+
+# enable pin controller
+CONFIG_PINCTRL=y
+
+# disable HTS221 trigger mode as the sensor's DRDY pin is not used on AZ3166
+CONFIG_HTS221_TRIGGER_NONE=y
```
Comment (erwango) [inline boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig:28]: Please move this to board's Kconfig.defconfig file, as done on similar boards.
(This being said, I'm a bit confused as I don't recall the rational)

**Review by erwango — DISMISSED** [review]
Thanks for contributing this board!
Two non blocking comments.

File: boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig (hunk 2 above)
Comment (kartben) [reply boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig:28]: Ah, good call!
This is actually to get rid of this kind of warning:

```

Comment (kartben) [reply boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig:28]: I will also move the default setting for SSD1306_REVERSE_MODE

File: boards/arm/az3166_iotdevkit/az3166_iotdevkit.dts (hunk 1 above)
Comment (kartben) [reply boards/arm/az3166_iotdevkit/az3166_iotdevkit.dts]: nice catch :) Fixed

**Comment by kartben** [comment]
> Thanks for contributing this board! Two non blocking comments.

Thanks for the super quick review :) I've addressed your two non-blocking-yet-very-valid comments in the latest version of the branch.

File: boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig (hunk 2 above)
Comment (erwango) [reply boards/arm/az3166_iotdevkit/az3166_iotdevkit_defconfig]: Great, now I remember :-)

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**